pkg/oc/cli/cmd/importimage.go

```diff
@@ -299,42 +298,28 @@ func (e importError) Error() string {
 }
 
 func (o *ImportImageOptions) waitForImport(resourceVersion string) (*imageapi.ImageStream, error) {
-	streamWatch, err := o.isClient.Watch(metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector("metadata.name", o.Name).String(), ResourceVersion: resourceVersion})
-	if err != nil {
-		return nil, err
-	}
-	defer streamWatch.Stop()
-
-	for {
-		select {
-		case event, ok := <-streamWatch.ResultChan():
-			if !ok {
-				return nil, fmt.Errorf("image stream watch ended prematurely")
+	var is *imageapi.ImageStream
+	err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
+		var err error
```

Comment:
don't shadow the outer-err ;-)

Reply:
this is what I took as a reference:
https://github.com/openshift/origin/blob/74a6a14d4f08eec96d890b663776225eae34d6cf/pkg/cmd/server/origin/openshift_apiserver.go#L675-L684

could you recommend what pattern should I rather follow? I would like to get the `is` out of the polled function